### PR TITLE
Make standard token compileable.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/1480

### CODE REVIEW

File: std/StandardToken.sol [hunk 1]
```diff
@@ -25,9 +33,9 @@ contract StandardToken is Token {
 	}
 
 	function transferFrom(address _from, address _to, uint256 _value) returns (bool success) {
-		if (allowance[_from][msg.sender] >= _value && balanceOf[_to] + _value >= balanceOf[_to]) {
-			allowance[_from][msg.sender] -= _value;
-			balanceOf[_to] += _value;
+		if (m_allowance[_from][msg.sender] >= _value && balance[_to] + _value >= balance[_to]) {
+			m_allowance[_from][msg.sender] -= _value;
+			balance[_to] += _value;
```
Comment (pirapira) [inline std/StandardToken.sol:42]: I think `balance[_from]` should be decreased.

**Review by pirapira — DISMISSED** [review]

**Comment by axic** [comment]
Why cannot we make public accessor fulfil the method signature requirement? I.e. `Token` defines `function balanceOf(address)` and `mapping (address => uint) balanceOf` should match that signature.

**Comment by chriseth** [comment]
@axic the interface requires an external and an internal implementation. The accessor function only provides the external implementation.

**Comment by pirapira** [comment]
I guess now it's between @axic and @chriseth .

**Comment by chriseth** [comment]
@axic can this be merged?

**Comment by axic** [comment]
> The accessor function only provides the external implementation.

I would have assumed it provides both the internal and the external interface. Isn't that a bug in that case?

**Comment by chriseth** [comment]
No, internally, you get a storage variable and not a function that returns the value of the storage variable.

**Comment by chriseth** [comment]
@axic Please note that this particular PR just fixes the token so that it compiles again. This is unrelated to the decision about whether or not public storage variables are considered implementors of public functions.

**Pull request merged**